test/transaction.ts

```diff
@@ -1702,6 +1702,53 @@ describe('Transaction', () => {
           ),
         );
       });
+
+      describe('when multiplexed session is enabled for read/write', () => {
+        before(() => {
+          process.env.GOOGLE_CLOUD_SPANNER_MULTIPLEXED_SESSIONS = 'true';
+          process.env.GOOGLE_CLOUD_SPANNER_MULTIPLEXED_SESSIONS_FOR_RW = 'true';
+        });
+        after(() => {
+          process.env.GOOGLE_CLOUD_SPANNER_MULTIPLEXED_SESSIONS = 'false';
+          process.env.GOOGLE_CLOUD_SPANNER_MULTIPLEXED_SESSIONS_FOR_RW =
+            'false';
+        });
+        it('should pass multiplexedSessionPreviousTransactionId in the BeginTransactionRequest upon retrying an aborted transaction', () => {
+          const fakePreviousTransactionId = 'fake-previous-transaction-id';
+          // multiplexed session
+          const multiplexedSession = Object.assign(
+            {multiplexed: true},
+            SESSION,
+          );
+          transaction = new Transaction(multiplexedSession);
+          // transaction option must contain the previous transaction id for multiplexed session
+          transaction.multiplexedSessionPreviousTransactionId =
+            fakePreviousTransactionId;
+          const stub = sandbox.stub(transaction, 'request');
+          transaction.begin();
+
+          const expectedOptions = {
+            isolationLevel: 0,
+            readWrite: {
+              multiplexedSessionPreviousTransactionId:
+                fakePreviousTransactionId,
+            },
+          };
+          const {client, method, reqOpts, headers} = stub.lastCall.args[0];
+
+          assert.strictEqual(client, 'SpannerClient');
+          assert.strictEqual(method, 'beginTransaction');
+          // request options should contain the multiplexedSessionPreviousTransactionId
+          assert.deepStrictEqual(reqOpts.options, expectedOptions);
```

Comment:
but asserting on begin if the request to begin transaction is actually containing the `multiplexedSessionPreviousTransactionId` set to the value